test/e2e/policyrecommendation_test.go

```diff
@@ -0,0 +1,256 @@
+
+	// Apply the recommended policies, and check the results
+	err = retrieveJobResult(t, data, jobId)
+	require.NoError(t, err)
+	cmd := fmt.Sprintf("kubectl apply -f %s", policyOutputYML)
+	_, stdout, stderr, err = data.RunCommandOnNode(controlPlaneNodeName(), cmd)
+	require.NoErrorf(t, err, "Error when running %v from %s: %v\nstdout:%s\nstderr:%s", cmd, controlPlaneNodeName(), err, stdout, stderr)
+	_, allPolicies, stderr, err := data.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("cat %s", policyOutputYML))
+	require.NoErrorf(t, err, "Error when running %v from %s: %v\nstdout:%s\nstderr:%s", cmd, controlPlaneNodeName(), err, stdout, stderr)
+
+	// Check recommended ANP counts
+	cmd = fmt.Sprintf("kubectl get anp -n %s", testNamespace)
+	_, stdout, stderr, err = data.RunCommandOnNode(controlPlaneNodeName(), cmd)
+	require.NoErrorf(t, err, "Error when running %v from %s: %v\nstdout:%s\nstderr:%s", cmd, controlPlaneNodeName(), err, stdout, stderr)
+	outputLines := strings.Split(stdout, "\n")
+	allowANPCnt := 0
+	rejectANPCnt := 0
+	for _, line := range outputLines {
+		if strings.Contains(line, "recommend-allow") {
+			allowANPCnt += 1
+		}
+		if strings.Contains(line, "recommend-reject") {
+			rejectANPCnt += 1
+		}
+	}
+	assert := assert.New(t)
+	assert.Equalf(expectedAllowANPCnt, allowANPCnt, fmt.Sprintf("Expected allow ANP count is: %d. Actual count is: %d. Recommended policies:\n%s", expectedAllowANPCnt, allowANPCnt, allPolicies))
+	assert.Equalf(expectedRejectANPCnt, rejectANPCnt, fmt.Sprintf("Expected reject ANP count is: %d. Actual count is: %d. Recommended policies:\n%s", expectedRejectANPCnt, rejectANPCnt, allPolicies))
+
+	// Check recommended ACNP counts
+	cmd = "kubectl get acnp"
+	_, stdout, stderr, err = data.RunCommandOnNode(controlPlaneNodeName(), cmd)
+	require.NoErrorf(t, err, "Error when running %v from %s: %v\nstdout:%s\nstderr:%s", cmd, controlPlaneNodeName(), err, stdout, stderr)
+	outputLines = strings.Split(stdout, "\n")
+	allowACNPCnt := 0
+	rejectACNPCnt := 0
+	for _, line := range outputLines {
+		if strings.Contains(line, "recommend-allow") {
+			allowACNPCnt += 1
+		}
+		if strings.Contains(line, "recommend-reject") {
+			rejectACNPCnt += 1
+		}
+	}
+	assert.Equalf(expectedAllowACNPCnt, allowACNPCnt, fmt.Sprintf("Expected allow ACNP count is: %d. Actual count is: %d. Recommended policies:\n%s", expectedAllowACNPCnt, allowACNPCnt, allPolicies))
+	assert.Equalf(expectedRejectACNPCnt, rejectACNPCnt, fmt.Sprintf("Expected reject ACNP count is: %d. Actual count is: %d. Recommended policies:\n%s", expectedRejectACNPCnt, rejectACNPCnt, allPolicies))
+}
+
+func runJob(t *testing.T, data *TestData) (stdout string, err error) {
+	cmd := "chmod +x ./theia"
+	rc, stdout, stderr, err := data.RunCommandOnNode(controlPlaneNodeName(), cmd)
+	if err != nil || rc != 0 {
+		return "", fmt.Errorf("Error when running %s from %s: %v\nstdout:%s\nstderr:%s", cmd, controlPlaneNodeName(), err, stdout, stderr)
```

Comment:
I think we don't capitalize the error string?

Reply:
fixed, thanks